handler.go

```diff
@@ -23,170 +26,159 @@ import (
 	"github.com/cloudradar-monitoring/cagent/pkg/monitoring/vmstat/types"
 )
 
-var ErrorTestWinUISettingsAreEmpty = errors.New("Please fill 'HUB URL', 'HUB USER' and 'HUB PASSWORD' from your Cloudradar account")
-
-func (ca *Cagent) initHubHTTPClient() {
-	if ca.hubHTTPClient == nil {
-		tr := *(http.DefaultTransport.(*http.Transport))
+func (ca *Cagent) initHubClientOnce() {
+	ca.hubClientOnce.Do(func() {
+		transport := &http.Transport{
+			ResponseHeaderTimeout: 15 * time.Second,
+		}
 		if ca.rootCAs != nil {
-			tr.TLSClientConfig = &tls.Config{RootCAs: ca.rootCAs}
+			transport.TLSClientConfig = &tls.Config{
+				RootCAs: ca.rootCAs,
+			}
 		}
-		if ca.Config.HubProxy != "" {
+		if len(ca.Config.HubProxy) > 0 {
+			// TODO(max): this should be moved to config sanitizer/validator layer.
```

Comment:
Do we have issue added to the tracking system for this?
I think leaving `TODO:` comments without clear plans when it will be fixed is not ok.

Reply:
> Do we have issue added to the tracking system for this?

I don't think so